site/content/contribute/more-info/webapp/unit-testing.md

```diff
@@ -10,48 +10,51 @@ aliases:
 
 ### Unit Tests for Component and Utility Files
 
-The last required step in building a web app component is to test it. {{<newtabref href="https://jestjs.io/en/" title="Jest">}} and {{<newtabref href="https://enzymejs.github.io/enzyme/" title="Enzyme">}} are the main frameworks/testing utilities used in unit testing components and utility files in our web app. Please visit their respective documentation for detailed information on how to get started, best practices, and updates. Enzyme is used to render and interact with React components in isolation, while Jest is used to perform snapshot testing against these components.
+The last required step in building a web app component is to test it. {{<newtabref href="https://jestjs.io/en/" title="Jest">}} and {{<newtabref href="https://testing-library.com/docs/react-testing-library/intro/" title="React Testing Library">}} are the main frameworks/testing utilities used in unit testing components and utility files in our web app. Please visit their respective documentation for detailed information on how to get started, best practices, and updates. React Testing Library is used to render and interact with React components in a way that closely mirrors how users interact with them in the browser, while Jest is used to perform snapshot testing against these components.
 
+With React Testing Library, you can focus on testing the behavior of your components, rather than their implementation details. This means that your tests will be more resilient to changes in your codebase, and will give you more confidence that your application is working as expected.
 If you need to unit test something related to Redux, please check out [Redux Unit and E2E Testing]({{<relref "/contribute/more-info/webapp/redux/testing.md">}}).
 
 #### Writing unit tests
 Below is a brief guide on how to do component testing:
 
-1. Match snapshots using default or expected props. Use shallow rendering to render just the component itself, and not any of its child components. Note that while using snapshots is convenient, do not rely solely on this for every test case, as changes can be easily overlooked when using the command `jest -updateSnapshot` to update multiple snapshots at once. For example:
+1. Use `render` from React Testing Library to render the component and its child components. Use `screen` from React Testing Library to interact with the rendered component and assert on the expected results. Match snapshots using default or expected props. Note that while using snapshots is convenient, do not rely solely on this for every test case, as changes can be easily overlooked when using the command `jest -updateSnapshot` to update multiple snapshots at once. For example:
```

Comment:
@nevyangelova We have our testing library helpers, https://github.com/mattermost/mattermost-server/blob/master/webapp/channels/src/tests/react_testing_utils.tsx.
And in most of the cases we will be using those instead of render from testing library.
It would be good if we mention that as well with an example.

Reply:
I agree. In addition, it would be great to consolidate importing of all "@testing-library/*" and custom renderer/queries/helpers in one place only (preferably at the same linked file as above) so that the consuming test file would import in one place only (easier to read and reference) like on how it's arranged here - https://testing-library.com/docs/react-testing-library/setup#custom-render. Something like:
```diff
// in [component].test.tsx
- import {screen} from '@testing-library/react';
- import userEvent from '@testing-library/user-event';
- import {renderWithIntl} from 'tests/react_testing_utils';
+ import {renderWithIntl, screen, userEvent} from 'tests/react_testing_utils';
```

Reply:
re: `jest -updateSnapshot`, either:
```
npm run test:updatesnapshot
```
or,
```diff
- jest -updateSnapshot
+ jest --updateSnapshot
```